### PR TITLE
[Core] Added a static method to convert OffsetDateTime to rfc1123 String

### PR DESCRIPTION
**Pull request by mssfang**
# Description
Fixes: https://github.com/Azure/azure-sdk-for-java/issues/24533

Added a static utility method to convert OffsetDateTime to rfc1123 String

# All SDK Contribution checklist:
- [X] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [X] Title of the pull request is clear and informative.
- [ X There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](../../CONTRIBUTING.md#building-and-unit-testing)
- [X] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/core/azure-core/src/main/java/com/azure/core/util/DateTimeRfc1123.java [hunk 1]
```diff
@@ -152,6 +152,65 @@ private static int parseMonth(final CharSequence date, final int beginIndex) {
...
+    private static void zeroPad(int value, StringBuilder sb) {
+        if (value < 10) {
+            sb.append("0");
+        }
+        sb.append(value);
+    }
+
     @Override
     public String toString() {
         return RFC1123_DATE_TIME_FORMATTER.format(this.dateTime);
```
Comment (alzimmermsft) [inline sdk/core/azure-core/src/main/java/com/azure/core/util/DateTimeRfc1123.java:216]: Have this use the static method

File: sdk/core/azure-core/src/main/java/com/azure/core/util/DateTimeRfc1123.java [hunk 2]
```diff
@@ -152,6 +152,65 @@ private static int parseMonth(final CharSequence date, final int beginIndex) {
...
+        StringBuilder sb = new StringBuilder(32);
+
+        switch (datetime.getDayOfWeek()) {
+            case MONDAY: sb.append("Mon, "); break;
+            case TUESDAY: sb.append("Tue, "); break;
+            case WEDNESDAY: sb.append("Wed, "); break;
+            case THURSDAY: sb.append("Thu, "); break;
+            case FRIDAY: sb.append("Fri, "); break;
+            case SATURDAY: sb.append("Sat, "); break;
+            case SUNDAY: sb.append("Sun, "); break;
```
Comment (alzimmermsft) [inline sdk/core/azure-core/src/main/java/com/azure/core/util/DateTimeRfc1123.java:174]: Missing a default case

File: sdk/core/azure-core/src/main/java/com/azure/core/util/DateTimeRfc1123.java [hunk 3]
```diff
@@ -152,6 +152,65 @@ private static int parseMonth(final CharSequence date, final int beginIndex) {
...
+            case MARCH: sb.append(" Mar "); break;
+            case APRIL: sb.append(" Apr "); break;
+            case MAY: sb.append(" May "); break;
+            case JUNE: sb.append(" Jun "); break;
+            case JULY: sb.append(" Jul "); break;
+            case AUGUST: sb.append(" Aug "); break;
+            case SEPTEMBER: sb.append(" Sep "); break;
+            case OCTOBER: sb.append(" Oct "); break;
+            case NOVEMBER: sb.append(" Nov "); break;
+            case DECEMBER: sb.append(" Dec "); break;
```
Comment (alzimmermsft) [inline sdk/core/azure-core/src/main/java/com/azure/core/util/DateTimeRfc1123.java:191]: Missing a default case

**Review by alzimmermsft — APPROVED** [review]

File: sdk/core/azure-core/src/main/java/com/azure/core/util/DateTimeRfc1123.java [hunk 4]
```diff
@@ -152,9 +147,72 @@ private static int parseMonth(final CharSequence date, final int beginIndex) {
         }
     }
 
+    /**
+     * Convert the {@link OffsetDateTime datetime} to datetime string in RFC1123 format.
+     *
+     * @param datetime The datetime string.
+     * @return The datetime string in RFC1123 format.
+     */
+    public static String toRFC1123String(OffsetDateTime datetime) {
```
Comment (srnagar) [inline sdk/core/azure-core/src/main/java/com/azure/core/util/DateTimeRfc1123.java:156]:
```suggestion
    public static String toRfc1123String(OffsetDateTime datetime) {
```

File: sdk/core/azure-core/src/test/java/com/azure/core/util/DateTimeRfc1123Tests.java [hunk 5]
```diff
@@ -53,4 +53,22 @@ public void parseDateTimeRfc1123StringWithIllegalArgumentException() {
...
+
+        //Random anytime
+        instantDate = OffsetDateTime.now();
+        dateString = RFC1123_DATE_TIME_FORMATTER.format(instantDate);
+        assertEquals(dateString, DateTimeRfc1123.toRFC1123String(instantDate));
+
+        // zero pad verified
+        String date1 = "Wed, 01 Dec 2021 01:01:01 GMT";
+        assertEquals(date1,
+            DateTimeRfc1123.toRFC1123String(OffsetDateTime.parse(date1, DateTimeFormatter.RFC_1123_DATE_TIME)));
```
Comment (srnagar) [inline sdk/core/azure-core/src/test/java/com/azure/core/util/DateTimeRfc1123Tests.java:72]: nit: consider using `@ParameterizedTest` and pass the OffsetDateTime and expected datetime strings to the test method.

File: sdk/core/azure-core/src/main/java/com/azure/core/util/DateTimeRfc1123.java [hunk 6]
```diff
@@ -152,9 +147,72 @@ private static int parseMonth(final CharSequence date, final int beginIndex) {
         }
     }
 
+    /**
+     * Convert the {@link OffsetDateTime datetime} to datetime string in RFC1123 format.
+     *
+     * @param datetime The datetime in OffsetDateTime format.
+     * @return The datetime string in RFC1123 format.
+     */
+    public static String toRfc1123String(OffsetDateTime datetime) {
```
Comment (srnagar) [inline sdk/core/azure-core/src/main/java/com/azure/core/util/DateTimeRfc1123.java:156]:
```suggestion
    public static String toRfc1123String(OffsetDateTime dateTime) {
```

**Review by srnagar — APPROVED** [review]

**Pull request merged**